File: 3rd Sem/Python/Programs/assignment-7/bank.py
class Bank:
    id = ""
    typeOfAccount = ""
    balance = ""
    transactionCount = 0

    def __init__(self, id, typeOfAccount, balance = 0):
        self.id = id
        self.typeOfAccount = typeOfAccount
        self.balance = balance
        Bank.transactionCount += 1

    def information(self):
        print("The account id of the customer is : {}".format(self.id))
        print("The type of account is : {}".format(self.typeOfAccount))
        print("The account balance is : {}".format(self.balance))

    def deposit(self, amount):
        self.balance = self.balance + amount
        print("The amount deposited is : {}".format(self.balance))

    def withdraw(self, amount):
        self.balance = self.balance - amount
        print("The withdrawn amount is : {}".format(self.balance))

    def display(self):
        print("The current balance is : {}".format(self.balance))
    
    def Transactions(self):
        print("No of transactions are : {}".format(Bank.transactionCount))

b1 = Bank(124, "Savings")
b1.information()
b1.deposit(1000)
b1.withdraw(100)
b1.display()
b1.Transactions()
print()
b2 = Bank(125, "FD")
b2.information()
b2.deposit(100000)
b2.display()
b2.Transactions()

File: 2nd Sem/C++/Programs/bank.cpp
#include <iostream>
#include <cstring>
using namespace std;

class Bank
{   
    public:
        char name[20];
        char account_no[20];
        char type[10];
        float bal;

        Bank()
        {
            bal = 0;
        }

        void deposit();
        void withdraw();
        void balance();
};

void Bank::deposit()
{
    float amount;

    cout << "Enter the amount to deposit: " << endl;
    cin >> amount;
    
    bal = bal + amount;
    cout << "Amount " << amount << " deposited" << endl;
}

void Bank::withdraw()
{
    float amount;

    cout << "Enter the amount to withdraw: " << endl;
    cin >> amount;

    if(bal < amount){
        cout << "Insufficient Balance.." << endl;
    }
    else
    {
        bal = bal - amount;
        cout << "Amount " << amount << " withdrawn" << endl;
    }
}

void Bank::balance()
{
    cout << "Balance = " << bal << endl;
}

int main()
{
    Bank customers[20];
    int n;
    cout << "Enter no of customers: " << endl;
    cin >> n;

    for (int i = 0; i < n; i++)
    {
        cout << "Enter customer " << i + 1 << " details: " << endl;
        cout << "Enter name: " << endl;
        cin >> customers[i].name;
        cout << "Enter account number: " << endl;
        cin >> customers[i].account_no;
        cout << "Enter account type: " << endl;
        cin >> customers[i].type;
    }

    int choice;
    char acc_num[20];

    while(1)
    {
        cout << "Enter what operation to be performed: " << endl;
        cout << "1. Deposit" << endl;
        cout << "2. Withdraw" << endl;
        cout << "3. Balance" << endl;
        cout << "4. Exit"<< endl;
        cin >> choice;
        
        if(choice == 0)
        {
            break;
        }

        cout << "Enter account no: " << endl;
        cin >> acc_num;

        switch(choice)
        {
            case 1:
                for (int i = 0; i < n; i++)
                {
                    if (!strcmp(acc_num, customers[i].account_no))
                    {
                        customers[i].deposit();
                        break;
                    }
                }
                break;
            case 2:
                for (int i = 0; i < n; i++)
                {
                    if (!strcmp(acc_num, customers[i].account_no))
                    {
                        customers[i].withdraw();
                        break;
                    }
                }
                break;
            case 3:
                for (int i = 0; i < n; i++)
                {
                    if (!strcmp(acc_num, customers[i].account_no))
                    {
                        customers[i].balance();
                        break;
                    }
                }
                break;
            case 4:exit(0);
            default: 
                cout << "Invalid option.." << endl;
        }
    }
}